numba/_helperlib.c

```diff
@@ -588,6 +588,25 @@ import_cython_function(const char *module_name, const char *function_name)
     return res;
 }
 
+NUMBA_EXPORT_FUNC(PyObject *)
+import_cython_function(PyObject *self, PyObject *args)
+{
+    const char *module_name;
+    const char *function_name;
+    void *p = NULL;
+    PyObject *res;
+
+    if (!PyArg_ParseTuple(args, "ss", &module_name, &function_name)) {
+	return NULL;
+    }
+    p = _import_cython_function(module_name, function_name);
+    if (p == NULL) {
+	return NULL;
```

Comment:
One complaint about an attribute error is that it becomes misleading for `cpdef` functions that operate on fused types (there are lots of these in `cython_special`).

If you have `capi.pyx`:

```cython
cimport cython

ctypedef fused floating:
    cython.double
    cython.float

cpdef api floating foo(floating x):
    return x
```

Then this will happen:

```python
>>> import capi
>>> hasattr(capi, "foo")
True
>>> capi.__pyx_capi__
{'__pyx_fuse_0foo': <capsule object "double (double, int __pyx_skip_dispatch)" at 0x104ddabd0>, '__pyx_fuse_1foo': <capsule object "float (float, int __pyx_skip_dispatch)" at 0x104ddaf60>}
```

On the Python level `foo` exists, but it isn't in the C API because of Cython's name mangling scheme. The roughly equivalent functionality in SciPy gives this error:

```python
>>> from scipy import LowLevelCallable
>>> LowLevelCallable.from_cython(capi, "foo")
Traceback (most recent call last):
  File "/Users/Josh/Projects/numba-dev/lib/python3.6/site-packages/scipy/_lib/_ccallback.py", line 126, in from_cython
    function = module.__pyx_capi__[name]
KeyError: 'foo'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/Josh/Projects/numba-dev/lib/python3.6/site-packages/scipy/_lib/_ccallback.py", line 130, in from_cython
    raise ValueError("No function {!r} found in __pyx_capi__ of the module".format(name))
ValueError: No function 'foo' found in __pyx_capi__ of the module
```

Thoughts about raising a similar `ValueError` (but swallowing the `KeyError`)?

Reply:
Ah yes, that is a noteworthy complaint, thanks for highlighting it. I agree with raising a `ValueError` as suggested as it would appear to suitably resolve the problem. Thanks.